fabric-tool-attributes-v1/src/main/java/net/fabricmc/fabric/api/tool/attribute/v1/ToolAttributeHolder.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.tool.attribute.v1;
+
+import com.google.common.collect.ImmutableSetMultimap;
+import com.google.common.collect.Multimap;
+
+import net.minecraft.entity.EquipmentSlot;
+import net.minecraft.entity.attribute.EntityAttributeModifier;
+import net.minecraft.item.ItemStack;
+
+/**
+ * Interface for adding various tool attributes to items.
+ */
+public interface ToolAttributeHolder {
+	Multimap<String, EntityAttributeModifier> EMPTY = ImmutableSetMultimap.of();
+
+	/**
+	 * @param stack The stack to check on.
+	 * @return The mining level of the item. 3 is equal to a diamond pick.
+	 */
+	int getMiningLevel(ItemStack stack);
+
+	/**
+	 * @param stack The stack to check on.
+	 * @return The mining speed of the item. 8.0 is equal to a diamond pick.
+	 */
+	float getMiningSpeed(ItemStack stack);
```

Comment:
Similar to the comment above, maybe a constants class for the default vanilla values.